modules/users/default.nix

```diff
@@ -245,10 +257,36 @@ in
             fi
 
             dscl . -create ${dsclUser} IsHidden ${if v.isHidden then "1" else "0"}
-            ${optionalString v.createHome "createhomedir -cu ${name}"}
           fi
-          # Always set the shell path, in case it was updated
-          dscl . -create ${dsclUser} UserShell ${lib.escapeShellArg (shellPath v.shell)}
+
+          # Always update all properties on users to keep them inline with configuration
+          dscl . -create ${dsclUser} PrimaryGroupID ${toString v.gid}
+          ${optionalString (v.description != null) "dscl . -create ${dsclUser} RealName ${lib.escapeShellArg v.description}"}
+          ${optionalString (v.shell != null) "dscl . -create ${dsclUser} UserShell ${lib.escapeShellArg (shellPath v.shell)}"}
+
+          # Don't allow updating home directory for current user and root
+          ${optionalString (v.home != null && v.name != "root") ''
+            if [[ ${name} == "$SUDO_USER" ]]; then
+              homeDirectory=$(dscl . -read ${dsclUser} NFSHomeDirectory)
+              homeDirectory=''${homeDirectory#NFSHomeDirectory: }
+              if [[ ${lib.escapeShellArg v.home} != "$homeDirectory" ]]; then
+                printf >&2 '\e[1;31merror: config contains the wrong home directory for %s, aborting activation\e[0m\n' ${name}
+                printf >&2 'nix-darwin does not support changing the home directory for the calling user (%s).\n' "$SUDO_USER"
+                printf >&2 '\n'
+                printf >&2 'Please set:\n'
+                printf >&2 '\n'
+                printf >&2 '    users.users.%s.home = "%s";\n' ${name} "$homeDirectory"
+                printf >&2 '\n'
+                printf >&2 'or remove it from your configuration.\n'
+                exit 1
+              fi
+            else
+              dscl . -create ${dsclUser} NFSHomeDirectory ${lib.escapeShellArg v.home}
+            fi
+          ''}
+          # This command doesn't do anything if the folder already exists which should be the case
+          # for $SUDO_USER and root, so it should be safe to run unconditionally.
+          ${optionalString v.createHome "createhomedir -cu ${name}"}
```

Comment:
I assume there’s no automated migration of files or anything, right? Do we want to be automating home directory changes, at least without a warning of some kind for this? We could split the difference by erroring out when the directory is wrong and printing commands to update it, which is an approach I like for avoiding automating “destructive” migrations while still making sure the activated configuration remains accurate.

Reply:
I've changed the PR to only run `createhomedir` on user creation (as `sysadminctl -addUser -home` won't create the home directory), I've removed the logic that updates `home` and added a check to ensure the `home` directory set in the config matches the system for all users.